In [1]:
import polars as pl 
import duckdb
import pandas as pd

In [2]:
chemin_parquet_demo = '../../parquet_demonstration.parquet'

In [3]:
categ_pl = pl.read_parquet('./categJuridique.parquet')
source_pl = pl.read_parquet(chemin_parquet_demo)

In [4]:
print(source_pl.columns)

['Nature', 'Fonction', 'LibCpte', 'ContNat', 'ArtSpe', 'ContFon', 'CodRD', 'MtBudgPrec', 'MtRARPrec', 'MtPropNouv', 'MtPrev', 'CredOuv', 'MtReal', 'MtRAR3112', 'OpBudg', 'MtSup', 'MtSup_1_Lib', 'MtSup_1_Val', 'MtSup_2_Lib', 'MtSup_2_Val', 'MtSup_3_Lib', 'MtSup_3_Val', 'MtSup_4_Lib', 'MtSup_4_Val', 'MtSup_5_Lib', 'MtSup_5_Val', 'MtSup_6_Lib', 'MtSup_6_Val', 'MtSup_7_Lib', 'MtSup_7_Val', 'MtSup_8_Lib', 'MtSup_8_Val', 'MtSup_9_Lib', 'MtSup_9_Val', 'MtSup_10_Lib', 'MtSup_10_Val', 'CaracSup_1_Lib', 'CaracSup_1_Val', 'CaracSup_2_Lib', 'CaracSup_2_Val', 'CaracSup_3_Lib', 'CaracSup_3_Val', 'CaracSup_4_Lib', 'CaracSup_4_Val', 'CaracSup_5_Lib', 'CaracSup_5_Val', 'CaracSup_6_Lib', 'CaracSup_6_Val', 'CaracSup_7_Lib', 'CaracSup_7_Val', 'TypOpBudg', 'OpeCpteTiers', 'DteStr', 'LibellePoste', 'IdPost', 'LibelleColl', 'IdColl', 'NatCEPL', 'Departement', 'Id_Fichier']


In [5]:
print(categ_pl.columns)

['siren', 'statutDiffusionUniteLegale', 'unitePurgeeUniteLegale', 'dateCreationUniteLegale', 'sigleUniteLegale', 'sexeUniteLegale', 'prenom1UniteLegale', 'prenom2UniteLegale', 'prenom3UniteLegale', 'prenom4UniteLegale', 'prenomUsuelUniteLegale', 'pseudonymeUniteLegale', 'identifiantAssociationUniteLegale', 'trancheEffectifsUniteLegale', 'anneeEffectifsUniteLegale', 'dateDernierTraitementUniteLegale', 'nombrePeriodesUniteLegale', 'categorieEntreprise', 'anneeCategorieEntreprise', 'dateDebut', 'etatAdministratifUniteLegale', 'nomUniteLegale', 'nomUsageUniteLegale', 'denominationUniteLegale', 'denominationUsuelle1UniteLegale', 'denominationUsuelle2UniteLegale', 'denominationUsuelle3UniteLegale', 'categorieJuridiqueUniteLegale', 'activitePrincipaleUniteLegale', 'nomenclatureActivitePrincipaleUniteLegale', 'nicSiegeUniteLegale', 'economieSocialeSolidaireUniteLegale', 'societeMissionUniteLegale', 'caractereEmployeurUniteLegale', 'categorieJuridiqueUniteLegale:1', 'libelleCategorieJuridique']

In [6]:
print(source_pl.shape)
print(categ_pl.shape)

(2683063, 60)
(168764, 36)


(2683063, 60)
(168764, 36)

In [7]:
categ_pl.head(5)

shape: (5, 36)
┌───────────┬───────────┬───────────┬───────────┬───┬───────────┬───────────┬───────────┬──────────┐
│ siren     ┆ statutDif ┆ unitePurg ┆ dateCreat ┆ … ┆ societeMi ┆ caractere ┆ categorie ┆ libelleC │
│ ---       ┆ fusionUni ┆ eeUniteLe ┆ ionUniteL ┆   ┆ ssionUnit ┆ Employeur ┆ Juridique ┆ ategorie │
│ str       ┆ teLegale  ┆ gale      ┆ egale     ┆   ┆ eLegale   ┆ UniteLega ┆ UniteLega ┆ Juridiqu │
│           ┆ ---       ┆ ---       ┆ ---       ┆   ┆ ---       ┆ le        ┆ le:1      ┆ e        │
│           ┆ str       ┆ bool      ┆ str       ┆   ┆ str       ┆ ---       ┆ ---       ┆ ---      │
│           ┆           ┆           ┆           ┆   ┆           ┆ str       ┆ i64       ┆ str      │
╞═══════════╪═══════════╪═══════════╪═══════════╪═══╪═══════════╪═══════════╪═══════════╪══════════╡
│ 130000029 ┆ O         ┆ null      ┆ 2005-11-0 ┆ … ┆ null      ┆ O         ┆ 7410      ┆ Groupeme │
│           ┆           ┆           ┆ 1         ┆   ┆           ┆           ┆           ┆ nt d'int │
│           ┆           ┆           ┆           ┆   ┆           ┆           ┆           ┆ érêt     │
│           ┆           ┆           ┆           ┆   ┆           ┆           ┆           ┆ public   │
│           ┆           ┆           ┆           ┆   ┆           ┆           ┆           ┆ (GIP…    │
│ 130000052 ┆ O         ┆ null      ┆ 2005-09-1 ┆ … ┆ null      ┆ N         ┆ 7410      ┆ Groupeme │
│           ┆           ┆           ┆ 6         ┆   ┆           ┆           ┆           ┆ nt d'int │
│           ┆           ┆           ┆           ┆   ┆           ┆           ┆           ┆ érêt     │
│           ┆           ┆           ┆           ┆   ┆           ┆           ┆           ┆ public   │
│           ┆           ┆           ┆           ┆   ┆           ┆           ┆           ┆ (GIP…    │
│ 130000060 ┆ O         ┆ null      ┆ 2005-10-1 ┆ … ┆ null      ┆ O         ┆ 7470      ┆ Groupeme │
│           ┆           ┆           ┆ 2         ┆   ┆           ┆           ┆           ┆ nt de    │
│           ┆           ┆           ┆           ┆   ┆           ┆           ┆           ┆ coopérat │
│           ┆           ┆           ┆           ┆   ┆           ┆           ┆           ┆ ion      │
│           ┆           ┆           ┆           ┆   ┆           ┆           ┆           ┆ sanita…  │
│ 130000078 ┆ O         ┆ null      ┆ 2004-04-1 ┆ … ┆ null      ┆ N         ┆ 7470      ┆ Groupeme │
│           ┆           ┆           ┆ 5         ┆   ┆           ┆           ┆           ┆ nt de    │
│           ┆           ┆           ┆           ┆   ┆           ┆           ┆           ┆ coopérat │
│           ┆           ┆           ┆           ┆   ┆           ┆           ┆           ┆ ion      │
│           ┆           ┆           ┆           ┆   ┆           ┆           ┆           ┆ sanita…  │
│ 130000086 ┆ O         ┆ null      ┆ 2005-10-1 ┆ … ┆ null      ┆ O         ┆ 7410      ┆ Groupeme │
│           ┆           ┆           ┆ 9         ┆   ┆           ┆           ┆           ┆ nt d'int │
│           ┆           ┆           ┆           ┆   ┆           ┆           ┆           ┆ érêt     │
│           ┆           ┆           ┆           ┆   ┆           ┆           ┆           ┆ public   │
│           ┆           ┆           ┆           ┆   ┆           ┆           ┆           ┆ (GIP…    │
└───────────┴───────────┴───────────┴───────────┴───┴───────────┴───────────┴───────────┴──────────┘

On a récupéré l'idcoll dans le parquet de démonstration, mais pas l'idpost (l'idcoll étant la même chose que idetab)

In [8]:
source_t1 = source_pl.filter(source_pl['IdColl'] == '20006779100018') #head est fait, ça existe

In [9]:
categ_t1 = categ_pl.filter(categ_pl['siren'] == '20006779100018') #C'est vide 

In [10]:
categ_t1

shape: (0, 36)
┌───────┬────────────┬────────────┬────────────┬───┬───────────┬───────────┬───────────┬───────────┐
│ siren ┆ statutDiff ┆ unitePurge ┆ dateCreati ┆ … ┆ societeMi ┆ caractere ┆ categorie ┆ libelleCa │
│ ---   ┆ usionUnite ┆ eUniteLega ┆ onUniteLeg ┆   ┆ ssionUnit ┆ Employeur ┆ Juridique ┆ tegorieJu │
│ str   ┆ Legale     ┆ le         ┆ ale        ┆   ┆ eLegale   ┆ UniteLega ┆ UniteLega ┆ ridique   │
│       ┆ ---        ┆ ---        ┆ ---        ┆   ┆ ---       ┆ le        ┆ le:1      ┆ ---       │
│       ┆ str        ┆ bool       ┆ str        ┆   ┆ str       ┆ ---       ┆ ---       ┆ str       │
│       ┆            ┆            ┆            ┆   ┆           ┆ str       ┆ i64       ┆           │
╞═══════╪════════════╪════════════╪════════════╪═══╪═══════════╪═══════════╪═══════════╪═══════════╡
└───────┴────────────┴────────────┴────────────┴───┴───────────┴───────────┴───────────┴───────────┘

In [11]:
source_t2 = source_pl.filter(source_pl['LibellePoste'] == 'TRESORERIE DE BRANTOME') #n'est pas vide

In [12]:
categ_t2 = categ_pl.filter(categ_pl['denominationUniteLegale'] == 'TRESORERIE DE BRANTOME')

In [13]:
categ_t2

shape: (0, 36)
┌───────┬────────────┬────────────┬────────────┬───┬───────────┬───────────┬───────────┬───────────┐
│ siren ┆ statutDiff ┆ unitePurge ┆ dateCreati ┆ … ┆ societeMi ┆ caractere ┆ categorie ┆ libelleCa │
│ ---   ┆ usionUnite ┆ eUniteLega ┆ onUniteLeg ┆   ┆ ssionUnit ┆ Employeur ┆ Juridique ┆ tegorieJu │
│ str   ┆ Legale     ┆ le         ┆ ale        ┆   ┆ eLegale   ┆ UniteLega ┆ UniteLega ┆ ridique   │
│       ┆ ---        ┆ ---        ┆ ---        ┆   ┆ ---       ┆ le        ┆ le:1      ┆ ---       │
│       ┆ str        ┆ bool       ┆ str        ┆   ┆ str       ┆ ---       ┆ ---       ┆ str       │
│       ┆            ┆            ┆            ┆   ┆           ┆ str       ┆ i64       ┆           │
╞═══════╪════════════╪════════════╪════════════╪═══╪═══════════╪═══════════╪═══════════╪═══════════╡
└───────┴────────────┴────────────┴────────────┴───┴───────────┴───────────┴───────────┴───────────┘

In [14]:
categ_pd = pd.read_parquet('./categJuridique.parquet')
source_pd = pd.read_parquet(chemin_parquet_demo)

In [15]:
categ_t2 = categ_pd[categ_pd.apply(lambda row: "20006779100018" in row.values, axis=1)] #Avec les deux ça me met des vents 
categ_t2

Empty DataFrame
Columns: [siren, statutDiffusionUniteLegale, unitePurgeeUniteLegale, dateCreationUniteLegale, sigleUniteLegale, sexeUniteLegale, prenom1UniteLegale, prenom2UniteLegale, prenom3UniteLegale, prenom4UniteLegale, prenomUsuelUniteLegale, pseudonymeUniteLegale, identifiantAssociationUniteLegale, trancheEffectifsUniteLegale, anneeEffectifsUniteLegale, dateDernierTraitementUniteLegale, nombrePeriodesUniteLegale, categorieEntreprise, anneeCategorieEntreprise, dateDebut, etatAdministratifUniteLegale, nomUniteLegale, nomUsageUniteLegale, denominationUniteLegale, denominationUsuelle1UniteLegale, denominationUsuelle2UniteLegale, denominationUsuelle3UniteLegale, categorieJuridiqueUniteLegale, activitePrincipaleUniteLegale, nomenclatureActivitePrincipaleUniteLegale, nicSiegeUniteLegale, economieSocialeSolidaireUniteLegale, societeMissionUniteLegale, caractereEmployeurUniteLegale, categorieJuridiqueUniteLegale:1, libelleCategorieJuridique]
Index: []

[0 rows x 36 columns]

In [16]:
categ_pd['denominationUniteLegale']

0                          GIP MAISON DE L'EMPLOI MEUSIENNE
1            GIP MEMOIRE DES PIERRES MEGALITHES DU MORBIHAN
2                                CITE SANITAIRE NAZAIRIENNE
3                        GPT COOP SANITAIRE INST REG CANCER
4                      GIP MAISON DEP PERSONNES HANDICAPEES
                                ...                        
168759                    L'HECTARE - TERRITOIRES VENDOMOIS
168760                             CAMPING MUNICIPAL ONDRES
168761    ORGANISME DE FONCIER SOLIDAIRE DE LA COMMUNE D...
168762    REGIE PUBLIQUE DE L'EAU ET DE L'ASSANISSEMENT ...
168763                  EPIC OFFICE DE TOURISME DE PRA LOUP
Name: denominationUniteLegale, Length: 168764, dtype: object

(2683063, 60)
(168764, 36)

In [17]:
conduck = duckdb.connect(database=':memory:', read_only=False)
source_duck = conduck.read_parquet('../../parquet_demonstration.parquet')
transco_t = conduck.read_parquet('./categJuridique.parquet')


In [18]:
requete_1 = ''' 
    SELECT siren FROM transco_t as t
    JOIN source_duck as s ON t.siren = s.IdColl'''
df_result1 = conduck.execute(requete_1).fetchdf()
#conduck.close()
print(df_result1.shape)
df_result1.head(5)

(0, 1)


Empty DataFrame
Columns: [siren]
Index: []

On va pré_découper 

In [19]:
requete_2 = ''' 
    SELECT DISTINCT 
        siren, 
        s.Idcoll
    FROM transco_t as t
    JOIN source_duck as s ON t.siren = LEFT(s.IdColl, 9)'''
df_result2 = conduck.execute(requete_2).fetchdf()
#conduck.close()
print(df_result2.shape)
df_result2.head(5)

(16721, 2)


siren          IdColl
0  265900084  26590008400038
1  211902903  21190290300015
2  241500230  24150023000016
3  214503385  21450338500012
4  216901231  21690123100011

In [20]:
requete_3 = '''
    SELECT DISTINCT siren
    FROM transco_t '''
df_requete_3 = conduck.execute(requete_3).fetchdf()
print(df_requete_3.shape)
df_requete_3.head(5)

(168764, 1)


siren
0  130000029
1  130000151
2  130000177
3  130000300
4  130000425

In [21]:
requete_4 = ''' 
    SELECT DISTINCT IdColl
    From source_duck'''
df_requete_4 = conduck.execute(requete_4).fetchdf()
print(df_requete_4.shape)
df_requete_4.head(5)

(16723, 1)


IdColl
0  20004962500011
1  21700152800017
2  21060039100010
3  21040203800017
4  21150122600016

In [22]:
requete_5 = ''' 
 SELECT DISTINCT IdColl 
 FROM source_duck as s
 ANTI JOIN transco_t as t ON t.siren = LEFT(s.IdColl, 9)'''
df_requete_5 = conduck.execute(requete_5).fetchdf()
print(df_requete_5.shape)
df_requete_5.head(5)

(2, 1)


IdColl
0  45408362700017
1  48068792000011

In [23]:
siren_unique_categ = 168764
siren_unique_fusion = 16721
siren_unique_source = 16723

Les deux idcoll qui ne sont pas dans categ juridique :  45408362700017 et  48068792000011

In [25]:
requete_6 = ''' 
    SELECT *
    FROM source_duck as s
    JOIN transco_t as t ON t.siren = LEFT(s.IdColl, 9)'''
df_result6 = conduck.execute(requete_6).fetchdf()
#conduck.close()
print(df_result6.shape)
df_result6.head(5)

(2682819, 96)


Nature Fonction LibCpte ContNat ArtSpe ContFon CodRD MtBudgPrec MtRARPrec  \
0  60632     None     nan     011  false    None     D   11962.00       nan   
1   6332     None     nan     012  false    None     D     450.00       nan   
2   6411     None     nan     012  false    None     D  365000.00       nan   
3  74834     None     nan     nan  false    None     R    4104.00       nan   
4   6331     None     nan     012  false    None     D    4250.00       nan   

  MtPropNouv  ... denominationUsuelle3UniteLegale  \
0   19335.00  ...                            None   
1     500.00  ...                            None   
2  352000.00  ...                            None   
3    4084.00  ...                            None   
4    5200.00  ...                            None   

  categorieJuridiqueUniteLegale activitePrincipaleUniteLegale  \
0                          7210                        84.11Z   
1                          7210                        84.11Z   
2                          7210                        84.11Z   
3                          7210                        84.11Z   
4                          7210                        84.11Z   

  nomenclatureActivitePrincipaleUniteLegale nicSiegeUniteLegale  \
0                                   NAFRev2                  19   
1                                   NAFRev2                  19   
2                                   NAFRev2                  19   
3                                   NAFRev2                  19   
4                                   NAFRev2                  19   

  economieSocialeSolidaireUniteLegale societeMissionUniteLegale  \
0                                   N                         N   
1                                   N                         N   
2                                   N                         N   
3                                   N                         N   
4                                   N                         N   

  caractereEmployeurUniteLegale categorieJuridiqueUniteLegale:1  \
0                             O                            7210   
1                             O                            7210   
2                             O                            7210   
3                             O                            7210   
4                             O                            7210   

     libelleCategorieJuridique  
0  Commune et commune nouvelle  
1  Commune et commune nouvelle  
2  Commune et commune nouvelle  
3  Commune et commune nouvelle  
4  Commune et commune nouvelle  

[5 rows x 96 columns]

In [28]:
df_result6[['LibelleColl', 'Nature', 'Fonction', 
            'LibCpte', 'MtReal', 'denominationUniteLegale', 
            'sigleUniteLegale', 'categorieJuridiqueUniteLegale', 'activitePrincipaleUniteLegale', 
            'nomenclatureActivitePrincipaleUniteLegale', 'nicSiegeUniteLegale', 'economieSocialeSolidaireUniteLegale', 
            'societeMissionUniteLegale', 'caractereEmployeurUniteLegale', 'categorieJuridiqueUniteLegale:1', 
            'libelleCategorieJuridique']]

LibelleColl Nature Fonction LibCpte MtReal  \
0                     AGONAC  60632     None     nan   None   
1                     AGONAC   6332     None     nan   None   
2                     AGONAC   6411     None     nan   None   
3                     AGONAC  74834     None     nan   None   
4                     AGONAC   6331     None     nan   None   
...                      ...    ...      ...     ...    ...   
2682814  COMMUNE DE BALIGNAC   7488     None    None   None   
2682815  COMMUNE DE BALIGNAC   7588     None    None   None   
2682816  COMMUNE DE BALIGNAC    761     None    None   None   
2682817  COMMUNE DE BALIGNAC    764     None    None   None   
2682818  COMMUNE DE BALIGNAC    774     None    None   None   

        denominationUniteLegale sigleUniteLegale  \
0             COMMUNE DE AGONAC             None   
1             COMMUNE DE AGONAC             None   
2             COMMUNE DE AGONAC             None   
3             COMMUNE DE AGONAC             None   
4             COMMUNE DE AGONAC             None   
...                         ...              ...   
2682814     COMMUNE DE BALIGNAC             None   
2682815     COMMUNE DE BALIGNAC             None   
2682816     COMMUNE DE BALIGNAC             None   
2682817     COMMUNE DE BALIGNAC             None   
2682818     COMMUNE DE BALIGNAC             None   

         categorieJuridiqueUniteLegale activitePrincipaleUniteLegale  \
0                                 7210                        84.11Z   
1                                 7210                        84.11Z   
2                                 7210                        84.11Z   
3                                 7210                        84.11Z   
4                                 7210                        84.11Z   
...                                ...                           ...   
2682814                           7210                        84.11Z   
2682815                           7210                        84.11Z   
2682816                           7210                        84.11Z   
2682817                           7210                        84.11Z   
2682818                           7210                        84.11Z   

        nomenclatureActivitePrincipaleUniteLegale  nicSiegeUniteLegale  \
0                                         NAFRev2                   19   
1                                         NAFRev2                   19   
2                                         NAFRev2                   19   
3                                         NAFRev2                   19   
4                                         NAFRev2                   19   
...                                           ...                  ...   
2682814                                   NAFRev2                   14   
2682815                                   NAFRev2                   14   
2682816                                   NAFRev2                   14   
2682817                                   NAFRev2                   14   
2682818                                   NAFRev2                   14   

        economieSocialeSolidaireUniteLegale societeMissionUniteLegale  \
0                                         N                         N   
1                                         N                         N   
2                                         N                         N   
3                                         N                         N   
4                                         N                         N   
...                                     ...                       ...   
2682814                                None                      None   
2682815                                None                      None   
2682816                                None                      None   
2682817                                None                      None   
2682818                                None                      None   

    

In [37]:
print( f"categorieJuridiqueUniteLegale uniques : \n {df_result6['categorieJuridiqueUniteLegale'].unique()} \n ")
print( f"nicSiegeUniteLegale uniques : \n {df_result6['nicSiegeUniteLegale'].unique()} \n  ")
print( f"economieSocialeSolidaireUniteLegale : \n {df_result6['economieSocialeSolidaireUniteLegale'].unique()}\n " )
print( f"caractereEmployeurUniteLegale : \n {df_result6['caractereEmployeurUniteLegale'].unique()}\n ")
print( f"categorieJuridiqueUniteLegale : \n {df_result6['categorieJuridiqueUniteLegale'].unique()}\n ")

categorieJuridiqueUniteLegale uniques : 
 [7210 7362 7361 7353 7348 7346 7355 7321 7323 7343 7344 7378 4140 7345
 7354 4150 7372 7220 7379 7367 7357 7229 7356 7373 7230 7340 7313 7351
 7410] 
 
nicSiegeUniteLegale uniques : 
 [  19   17   12   14   10   25   11   38   18   13   15   16   21   74
   41  267   54  193   69  244  146   33  136  117  110  179  167   55
   63   34  145  109   24  128  143   45  115   67   65  107  124  222
   23  190   22   53  152   79  174  138   70  242   26   73 1787  106
   78  140   64   71   31   57  144   77  423  175  201   30   20   81
   84   86   90   66   91  236   49   60  156  166  123  218  113   75
   80  540  129  134   92   72   48   39   37   28  151   27  198   32
  239  425  158  111   58   46   95   35   89   29  173   36  402   51
  231  127   76  103   93  159   85  362  101  785  112   42 1453  170
  232  658  197  164  187   56  209  105  102   40  312  130  260   43
  217  989   62  317   99  221  163  738   59  182  275  210  12